### ISSUE TITLE
I_Configuration and I_ConfigurationElement default implementation

### ISSUE THREAD

**Issue by SgtFlame** [indie-zen/zen-plugin#1]
Zen Plugins need a way to be configured.  In 1.x, we used XML, but in 2.0 we want to avoid requiring an XML parser.

The current default implementation (XMLConfiguration / XMLConfigurationElement) require libxml.  We need a new implementation that can be used in place of XMLConfiguration.  This new implementation should be able to be driven by a configuration parser as well as support embedded static configurations in compiled C++ code.

Possibly keep the XML implementation with a conditional that indicates if it should be compiled or not, possibly driven by CMake determining if libxml is available.

**Comment by hatboyzero** [comment]
Would a JSON based approach be acceptable?  I think I had actually implemented a JSON based plugin configuration scheme using Zen 1.x or a close approximation of it on a past project...

**Comment by SgtFlame** [comment]
JSON would be awesome; maybe two different implementations (JSON / XML) of the same interfaces in two separate libraries that get statically linked?

**Comment by hatboyzero** [comment]
Yeah, that would be simple enough - if I get some time, I'll throw together a basic impl if I can't find my original one to reference.  Working exclusively on Mac OS platforms lately, so the resulting CMake scripts might need some cross-platform love...
